### ISSUE TITLE
Implement new Concept Exercise: threads

### ISSUE THREAD

**Issue by workingjubilee** [exercism/rust#1069]
This issue describes how to implement the `threads` concept exercise for the rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Learning to write truly good multithreaded code is an enormous undertaking. Performing basic parallelism is not. The aim of this exercise is to provide a simple understanding of Rust's promised [fearless concurrency](https://doc.rust-lang.org/book/ch16-00-concurrency.html).

## Learning objectives

- be able to use [std::thread] to do simple parallel processing
- know how to explicitly use JoinHandles

## Out of scope

- This will likely be the first core concurrency exercise so it is advised to avoid _anything_ else that is not needed yet. The absolute minimum of sync primitives should be used, and ideally none (e.g. iterating over the JoinHandles wrapped <T>).

## Concepts

- multithreading
- [thread::spawn](https://doc.rust-lang.org/std/thread/fn.spawn.html)
- [JoinHandle<T>]

## Prerequisites

This will require a conceptual understanding of
- complex types and methods
- handling Result/Option types
- loops and iteration
- ownership and lifetimes
- specifically, threads in practice require `move` to do parallel work
- and practical implementations often require explicit copying/cloning
- destructors, probably
- panics (threads do that a lot)

...baaasically this comes after everything else.

### Hints

- [std::thread]
- [Fearless Concurrency]

### After

- [std::thread]
- [Fearless Concurrency]
- [Rust By Example: Map Reduce](https://doc.rust-lang.org/stable/rust-by-example/std_misc/threads/testcase_mapreduce.html)

## Representer

No changes required

## Analyzer

No changes required

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.  In addition you may want to make use of the shell script while in the `concept` directory, with the argument of the exercise name, e.g. `sh boil_exercise.sh threads`

Just explicitly copying "Parallel Letter Frequency" is probably a little lazy, but if you did but provided more of a guided tour of threading in doing so, it would not be entirely inappropriate.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[std::thread]: https://doc.rust-lang.org/std/thread/
[JoinHandle<T>]: https://doc.rust-lang.org/std/thread/struct.JoinHandle.html
[Fearless Concurrency]: https://doc.rust-lang.org/book/ch16-00-concurrency.html

**Comment by senekor** [comment]
After several failed attempts to create a high-quality syllabus, I believe any future attempts will have to do their own design work from scratch.
